### DISCUSSION TITLE
Fatal error: Uncaught GuzzleHttp\Exception\ClientException: Client error:

### DISCUSSION THREAD

**Discussion by marten66** [kreait/firebase-php#564] [Q&A]
I got this error, It worked before. I have been getting this error since about February 1, 2021. 

Fatal error: Uncaught GuzzleHttp\Exception\ClientException: Client error: `POST https://fcm.googleapis.com/v1/projects/veganesingles/messages:send` resulted in a `404 Not Found` response: (truncated...) in D:\wwwroot\veganesingles\www\vendor\guzzlehttp\guzzle\src\Exception\RequestException.php:113 Stack trace: #0 D:\wwwroot\veganesingles\www\vendor\guzzlehttp\guzzle\src\Middleware.php(69): GuzzleHttp\Exception\RequestException::create(Object(GuzzleHttp\Psr7\Request), Object(GuzzleHttp\Psr7\Response), NULL, Array, NULL) #1 D:\wwwroot\veganesingles\www\vendor\guzzlehttp\promises\src\Promise.php(204): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Response)) #2 D:\wwwroot\veganesingles\www\vendor\guzzlehttp\promises\src\Promise.php(153): GuzzleHttp\Promise\Promise::callHandler(1, Object(GuzzleHttp\Psr7\Response), NULL) #3 D:\wwwroot\veganesingles\www\vendor\guzzlehttp\promises\src\TaskQueue.php(48): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}() #4 D:\wwwroot\veganesingles\www\vendor\guzzleh in D:\wwwroot\veganesingles\www\vendor\kreait\firebase-php\src\Firebase\Exception\Messaging\NotFound.php on line 22

My Code: 
`<?php

/**
31.01.2021
wird includiert von inc/function/message.php
*/

// for standalone testing
require_once("../../inc/conf.php");

//require '../vendor/autoload.php';
require "$rootpath/www/vendor/autoload.php";

use Kreait\Firebase\Factory;
use Kreait\Firebase\Messaging\CloudMessage;
use Kreait\Firebase\Messaging\Message;

$factory = (new Factory())
    ->withServiceAccount("$rootpath/firebase/firebase_credentials.json")
    ->withDatabaseUri('https://veganesingles-default-rtdb.firebaseio.com')
;

$messaging = $factory->createMessaging();    

// for testing
if(!$deviceToken) {
$deviceToken = "123";  
$title = "veganesingles";
$body = "Test Nachricht...";
}
 

 $notification = [
        "body" => $body,
        "title" => $title
    ];

$data = [
        "type" => "Nachricht",
        "site" => "tablinks/message"
    ];   

//echo $deviceToken;
//print_r($notification);

$message = CloudMessage::withTarget('token', $deviceToken)
    ->withNotification($notification) // optional
    //->withData($data) // optional
;

$messaging->send($message);

?>

`

**Comment by jeromegamez** [comment]
This looks like a registration token that has either been unregistered or is not registered to the same Firebase project as the Admin SDK.

The official docs go into more details why a token can be unregistered: https://firebase.google.com/docs/cloud-messaging/http-server-ref?hl=en#error-codes

PS: Please use Markdown formatting for code/error Blocks, they make them much easier to read 😅

**Reply by marten66** [reply]
I got en error code 404. There is no error code 404 there in that documentation. URL should be fine.
i generated a new firebase_credentials.json file here. still the same problem. Google firebase  ist so complicate! A lot of pages and services and so on... puhh. 

![grafik](https://user-images.githubusercontent.com/60956844/107114262-cf5e1580-6864-11eb-83db-49efefe108d3.png)

**Reply by jeromegamez** [reply]
Yes it is 😂, and I realize that the link to the docs was not helpful here 🙈. But here, I can tell you from experience and from previous reports that it's most probably the registration token you're sending the message to (target), these are the most likely reasons:

- The target is registered with another Firebase project that you're sending it from. So, for example, an app is using Project A, but the Admin SDK is using Project B.
- The token has been unregistered: on a mobile app this can happen when a user uninstalls/reinstalls the app (and receives a new registration token). In the browser it can happen when the browser gets updated. It can also happen when a user logs out of the app.
- Perhaps (I'm not sure here) FCM must be explicitly enabled I the Google Cloud Project API console. Perhaps FCM can't find the project at all 🤔

Try it with another token, preferably a newly created one, this should take care of the first two cases 🤞

**Reply by marten66** [reply]
You were right.  The problem was the invalid device token. The app did not update the device token on the server after login.  I fixed the bug on the API, now it works. Thank you!

**Reply by marten66** [reply]
Is it possible to suppress this error message?  If the token does not exist, there is an error in my app and Webapp. That shouldn't be.  My process should work anyway, regardless of whether the push message could be delivered or not.

**Reply by jeromegamez** [reply]
You can suppress the error by wrapping the call in a try/catch block

```php
try {
    $messaging->send($message);
} catch (\Kreait\Firebase\Exception\Messaging\NotFound $e) {
    // Do nothing
}

**Reply by rob-gould-implerus** [reply]
I am also getting this Guzzle error: 404 - - - and I see why, but I'm not sure how to fix it.

[php7:error] [pid 25585] [client 108.28.118.150:53124] PHP Fatal error:  Uncaught GuzzleHttp\\Exception\\ClientException: Client error: `GET https://oscal-2021.firebaseio.com/table_ac-2_smt_c_0/-MTHcvF1my3n8TtxsJch/checkpoint/o/0` resulted in a `404 Not Found`

The issue is that the PHP library is trying to pull from https://oscal-2021.firebaseio.com, but when I log into Firebase, it's saying the url should instead be : https://oscal-2021-default-rtdb.firebaseio.com/

In my config in my client, I've got:

const firebaseConfig = {
  authDomain: "oscal-2021.firebaseapp.com",
  databaseURL: "https://oscal-2021-default-rtdb.firebaseio.com",

I re-generated a new key but it doesn't seem to make any difference.  What should I be doing differently?

**Reply by jeromegamez** [reply]
For specifying the database url, please see https://github.com/kreait/firebase-php/discussions/558#discussioncomment-319193
